csharp/src/Ice/Communicator.cs

```diff
@@ -1337,6 +1383,60 @@ private void AddAllAdminFacets()
             return null;
         }
 
+        // Make sure that all assemblies that are referenced by this process are actually loaded. This is necessary so
+        // we can use reflection on any type in any assembly because the type we are after will most likely not be in
+        // the current assembly and, worse, may be in an assembly that has not been loaded yet. (Type.GetType() is no
+        // good because it looks only in the calling object's assembly and mscorlib.dll.)
+        private static void LoadAssemblies()
+        {
+            Assembly[] assemblies = AppDomain.CurrentDomain.GetAssemblies();
+            var newAssemblies = new List<Assembly>();
+            foreach (Assembly assembly in assemblies)
+            {
+                if (!_loadedAssemblies.ContainsKey(assembly.FullName!))
```

Comment:
Should we lock _staticLock (which should be named _staticMutex) when reading/writing _loadedAssemblies?

Reply:
Yes I will fix